widgets/AttributesTable/_GraphicsMixin.js

```diff
@@ -648,15 +648,45 @@ define([
             this.clearGraphicsLayer(this.bufferGraphics);
         },
 
-        clearGraphicsLayer: function (layer) {
+        clearGraphicsLayer: function (layer, specificFeatures) {
             if (layer) {
-                layer.clear();
+                if (specificFeatures && specificFeatures.selection && specificFeatures.idProperty) {
+                    this.clearSpecificFeatures(layer, specificFeatures);
+                } else {
+                    layer.clear();
+                }
             }
             this.setToolbarButtons();
             topic.publish(this.attributesContainerID + '/tableUpdated', this);
 
         },
 
+        clearSpecificFeatures: function (layer, specificFeatures) {
```

Comment:
this method should be named `clearSpecificGraphics` since it is not specific to Features and can be used for any of the graphic layers.